gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -80,6 +82,14 @@
 
   public abstract ApiCallContext getDefaultCallContext();
 
+  @BetaApi("The surface for streaming is not stable yet and may change in the future.")
+  @Nullable
+  public abstract Watchdog getWatchdog();
```

Comment:
It seems like in the context of `ClientContext`, the term "watchdog" is unclear - maybe "stream watchdog" would make more sense?